linkml/generators/pydanticgen/pydanticgen.py

```diff
@@ -775,6 +1082,38 @@ def _subclasses(cls: Type):
 _TEMPLATE_NAMES = sorted(list(set([c.template for c in _subclasses(TemplateModel)])))
 
 
+def _import_to_path(module: str) -> Path:
+    """Make a (relative) ``Path`` object from a python module import string"""
+    # handle leading .'s separately..
+    _, dots, module = re.split(r"(^\.*)(?=\w)", module, maxsplit=1)
+    # treat zero or one dots as a relative import to the current directory
+    dir_pieces = ["../" for _ in range(max(len(dots) - 1, 0))]
+    dir_pieces.extend(module.split("."))
+    dir_pieces[-1] = dir_pieces[-1] + ".py"
+    return Path(*dir_pieces)
+
+
+def _ensure_inits(paths: List[Path]):
+    """For a set of paths, find the common root and it and all the subdirectories have an __init__.py"""
+    # if there is only one file, there is no relative importing to be done
+    if len(paths) <= 1:
+        return
+    common_path = Path(os.path.commonpath(paths))
+
+    if not (ipath := (common_path / "__init__.py")).exists():
```

Comment:
Wow! A walrus operator spotted in the wild!

Reply:
I WAS SO EXCITED

Reply:
The walrus even respects the parens scope which blew my mind